R/detect_mistakes.R

```diff
@@ -236,33 +237,41 @@ detect_mistakes <- function(user,
   }
 
 
-  # It is now safe to call call_standardise_formals on student code
-  user <- call_standardise_formals(user, env = env)
-  user_names <- real_names(user)
-
   # 5. Check that every named argument in the solution appears in the user code.
   #    The outcome of this order is that when a user writes na = TRUE, gradethis
   #    will tell them that it expected an na.rm argument, not that na is a surplus
   #    argument.
-  solution_names <-  real_names(solution) # original solution_names was modified above
-  missing_args <- solution_names[!(solution_names %in% user_names)]
+ 
+  explicit_solution <-call_standardise_formals(unpipe_all(answer),env = env,include_defaults = FALSE)
```

Comment:
```suggestion
  explicit_solution <- call_standardise_formals(
    unpipe_all(solution), 
    env = env, 
    include_defaults = FALSE
  )
```

Reply:
humm, I think we can't do this like that because of line 59 we do :

```
  if (is.call(solution)) {
    solution <- call_standardise_formals(unpipe_all(solution), env = env)
  }
```

so we cant use ` include_defaults = FALSE`  on an already expanded call

Reply:
Ah! Thank you.

Can we rename `answer` to `solution_original`? 🙌